src/Http/Cookie/CookieCollection.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @since         3.5.0
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Http\Cookie;
+
+use ArrayIterator;
+use InvalidArgumentException;
+use IteratorAggregate;
+
+/**
+ * Cookie Collection
+ */
+class CookieCollection implements IteratorAggregate
```

Comment:
Is there any opportunity to repurpose/extend/reuse the `CookieCollection` in `Cake\Http\Client`?